oauthlib/oauth2/rfc6749/endpoints/authorization.py

```diff
@@ -64,7 +64,12 @@ class AuthorizationEndpoint(BaseEndpoint):
     def __init__(self, default_response_type, default_token_type,
                  response_types):
         BaseEndpoint.__init__(self)
-        self._response_types = response_types
+        self._response_types = {}
+        # response_types are sorted internally so ordered comparison is faster/easier later
+
+        for k, v in response_types.iteritems() if sys.version_info[0] == 2 else iter(response_types.items()):
```

Comment:
This reads really ugly. Can we just use `response_types.items()` for both Python 2 and Python 3? I know it's slightly less memory-efficient in Py2, but `response_types` is going to be pretty darn small, so I doubt it matters.

Failing that, I think we should use [`six`](https://pythonhosted.org/six/) to support multi-version Python, instead of scattering compatibility code throughout our code base.